src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs

```diff
@@ -76,15 +79,24 @@ public async Task StartAsync(CancellationToken cancellationToken = default)
             _logger.Started();
         }
 
-        private async Task HandleBackgroundException(BackgroundService backgroundService)
+        private async Task TryStartBackgroundServiceAsync(
+            BackgroundService backgroundService, CancellationToken cancellationToken)
         {
             try
             {
+                // Fire IHostedService.Start
+                await backgroundService.StartAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
So this behavior applies to both StartAsync and StopAsync. I guess that makes sense.

Reply:
It applies to both `StartAsync` and the await of the `ExecuteTask`, this ensures that if `StartAsync` was to throw immediately - it is correctly handled.